bin/textcleaner

```diff
@@ -0,0 +1,619 @@
+#!/bin/bash
+# 
+# Developed by Fred Weinhaus 6/9/2009 .......... revised 5/12/2020
+# 
+# ------------------------------------------------------------------------------
+# 
+# Licensing:
+# 
+# Copyright � Fred Weinhaus
+# 
+# My scripts are available free of charge for non-commercial use, ONLY.
+# 
+# For use of my scripts in commercial (for-profit) environments or 
+# non-free applications, please contact me (Fred Weinhaus) for 
+# licensing arrangements. My email address is fmw at alink dot net.
+# 
+# If you: 1) redistribute, 2) incorporate any of these scripts into other 
+# free applications or 3) reprogram them in another scripting language, 
+# then you must contact me for permission, especially if the result might 
+# be used in a commercial or for-profit environment.
+# 
+# My scripts are also subject, in a subordinate manner, to the ImageMagick 
+# license, which can be found at: http://www.imagemagick.org/script/license.php
+# 
+# ------------------------------------------------------------------------------
+# 
+####
+# 
+# USAGE: textcleaner [-r rotate] [-l layout] [-c cropoff] [-g] [-e enhance ] 
+# [-f filtersize] [-o offset] [-u]  [-t threshold] [-s sharpamt] [-s saturation] 
+# [-a adaptblur] [-T] [-p padamt] [-b bgcolor] [-F fuzzval] [-i invert] infile outfile
+# USAGE: textcleaner [-help]
+# 
+# OPTIONS:
+# 
+# -r	  rotate			rotate image 90 degrees in direction specified if 
+#                           aspect ratio does not match layout; options are cw 
+#                           (or clockwise), ccw (or counterclockwise) and n 
+#                           (or none); default=none or no rotation
+# -l      layout            desired layout; options are p (or portrait) or 
+#                           l (or landscape); default=portrait
+# -c      cropoff			image cropping offsets after potential rotate 90; 
+#                           choices: one, two or four non-negative integer comma 
+#                           separated values; one value will crop all around; 
+#                           two values will crop at left/right,top/bottom; 
+#                           four values will crop left,top,right,bottom
+# -g                        convert document to grayscale before enhancing
+# -e      enhance           enhance image brightness before cleaning;
+#                           choices are: none, stretch or normalize; 
+#                           default=stretch
+# -f      filtersize        size of filter used to clean background;
+#                           integer>0; default=15
+# -o      offset            offset of filter in percent used to reduce noise;
+#                           integer>=0; default=5
+# -u                        unrotate image; cannot unrotate more than 
+#                           about 5 degrees
+# -t      threshold			text smoothing threshold; 0<=threshold<=100; 
```

Comment:
It looks like the textcleaner script has builtin support for the threshold option. Is there a reason for performing it as an separate additional step?